midend/predication.cpp

```diff
@@ -199,11 +224,17 @@ const IR::Node* Predication::preorder(IR::AssignmentStatement* statement) {
             lvalue_name(liveAssigns.back()->left) == lvalue_name(statement->left)) {
             liveAssigns.pop_back();
         }
-        auto rightStatement = clone(statement->right)->apply(replacer);
-        liveAssignments[statementName] =
-            new IR::AssignmentStatement(statement->left, rightStatement);
-        liveAssigns.push_back(liveAssignments[statementName]);
     }
+
+    // Push statement which now contains a Mux expression in the statement->right.
+    // ExpressionReplacer is applied here which takes care of correct transforming.
+    liveAssignments[statementName] =
+        new IR::AssignmentStatement(statement->left, clone(statement->right)->apply(replacer));
+    liveAssigns.push_back(liveAssignments[statementName]);
+    LOG1("Finished visiting statement");
```

Comment:
In general we reserve level 1 for more important messages, perhaps you can make this into a higher level.